java/Mekhmet-Efe_Adzhar/src/com/zoolatech/lecture10/tasks/_1/FileRedactor.java

```diff
@@ -0,0 +1,53 @@
+package com.zoolatech.lecture10.tasks._1;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Arrays;
+
+final class FileRedactor {
+
+    private final File file;
+
+    public FileRedactor(File file) {
+        this.file = file;
+    }
+
+    public void showFiles() {
+        File[] files = file.listFiles();
+        int counter = 0;
+
+        assert files != null;
+        for (File value : files) {
+            if (value.isFile()) {
+                System.out.println("File number " + ++counter + " : " + value.getName());
+            }
+        }
+    }
+
+    public boolean generateFiles(String fileName) throws IOException {
+        File[] files = file.listFiles();
+        assert files != null;
```

Comment:
it's better to print some error message to System.err stream instead of assert. To be honest it's vary rarely used in production code